### PR TITLE
Change explicit bit fields to a packed 128-bit int for the short pointers. 

### PR DESCRIPTION
**Pull request by ujjwal-shekhar**
- Removed the 14 short pointers, each of size `SHORT_DELTA` bits; instead.
- There are now 2 128-bit integers that have 7 packed `SHORT_DELTA` bits.
- Added an unsigned `num_occupied` to directly get the total occupancy of a chunk. 

## Summary by CodeRabbit

- **Bug Fixes**
	- Adjusted benchmark tests for tree structures to disable certain depth tests (depths 6, 7, and 8).
	- Updated correctness tests for tree structures to modify depth limits, impacting node levels during execution.
- **New Features**
	- Enhanced internal representation of tree nodes for improved memory efficiency and management.
	- Added tracking for the number of occupied short delta pointers in tree structures.
- **Documentation**
	- Improved output visibility during testing by commenting out console outputs in traversal functions.

### CODE REVIEW

**Comment by renau** [comment]
There are conflicts that need to be solved before merging.

File: hhds/tree.hpp [hunk 1]
```diff
@@ -375,6 +311,8 @@ class tree {
                 pointers_stack[parent_chunk_id].set_last_child_s_at(offset, INVALID);
             }
         }
+        // Decrement the number of occupied slots in the chunk
+        pointers_stack[parent_chunk_id].set_num_occupied(parent_chunk_offset - 1);
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:315]: This part of the code is after the section with:

```cpp
if ((parent_id & CHUNK_MASK) == 0) {
            //......
            return parent_id;
        }
```

Thus, the `const auto parent_chunk_offset = (parent_id & CHUNK_MASK)` cannot be equal to zero. This eliminates the need of adding an unnecessary check.

File: hhds/tree.hpp [hunk 2]
```diff
@@ -1169,15 +1059,17 @@ Tree_pos tree<X>::get_sibling_prev(const Tree_pos& sibling_id) {
...
+          : static_cast<Tree_pos>((prev_sibling << CHUNK_SHIFT) 
+                                  + pointers_stack[prev_sibling].get_num_occupied());
+    // // TODOTODO
+    // if (prev_sibling != INVALID) {
+    //     for (short offset = NUM_SHORT_DEL - 1; offset >= 0; offset--) {
+    //         if (_contains_data(prev_sibling + offset)) {
+    //             return static_cast<Tree_pos>(prev_sibling + offset);
+    //         }
+    //     }
+    // }
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:1072]: This was old code that is not needed anymore. Will be removed in the next commit.

File: hhds/tree.hpp [hunk 3]
```diff
@@ -305,10 +243,8 @@ class tree {
     }
 
     Tree_pos _try_fit_child_ptr(const Tree_pos &parent_id, const Tree_pos &child_id) {
-        // Check and throw accordingly
-        // if (!_check_idx_exists(parent_id) || !_check_idx_exists(child_id)) {
-        //     throw std::out_of_range("_try_fit_child_ptr: Index out of range");
-        // }
+        I(_check_idx_exists(parent_id), "First parent_id index out of range");
+        I(_check_idx_exists(child_id), "child_id index out of range");
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:247]: Fixed. Will be available in the next commit.

File: hhds/tree.hpp [hunk 4]
```diff
@@ -21,6 +21,8 @@
 #include <iterator>
 #include <cstddef> 
 
+#include "iassert.hpp"
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:24]: The dependency is handled by Bazel.

File: hhds/tree.hpp [hunk 5]
```diff
@@ -237,22 +182,18 @@ class tree {
...
         // idx >= 0 not needed for unsigned int
-        return idx < static_cast<Tree_pos>(data_stack.size());
+        return idx < static_cast<Tree_pos>(pointers_stack.size() << CHUNK_SHIFT);
     }
-    bool _contains_data(const Tree_pos &idx) const noexcept {
-        return (idx < data_stack.size() && data_stack[idx].has_value());
+
+    inline bool _contains_data(const Tree_pos &idx) const noexcept {
+        return (pointers_stack[idx >> CHUNK_SHIFT].get_num_occupied() >= (idx & CHUNK_MASK));
+        // return (idx < data_stack.size() && data_stack[idx].has_value());
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:192]: The name of the variable has been changed to `num_short_del_occ` to clarify the intent. The usage is perfectly fine. Will be available in the next commit.

File: hhds/tree.hpp [hunk 6]
```diff
@@ -1218,6 +1110,11 @@ Tree_pos tree<X>::append_sibling(const Tree_pos& sibling_id, const X& data) {
         _update_parent_pointer(first_sib, new_parent_id);
     }
 
+    // Increment the number of occupied slots in the sibling chunk
+    pointers_stack[new_sib >> CHUNK_SHIFT].set_num_occupied(
+                                                ((unsigned)new_sib & CHUNK_MASK)
+                                            );
+
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:1117]: The current code is correct, and requires no changes.

File: hhds/tree.hpp [hunk 7]
```diff
@@ -1315,8 +1216,12 @@ Tree_pos tree<X>::add_child(const Tree_pos& parent_index, const X& data) {
     const auto child_chunk_id = _create_space(data);
     const auto new_parent_id = _try_fit_child_ptr(parent_index, 
                                                   child_chunk_id << CHUNK_SHIFT);
+
     pointers_stack[child_chunk_id].set_parent(new_parent_id);
 
+    // Set num occupied to 0
+    pointers_stack[child_chunk_id].set_num_occupied(0);
+
```
[the remark this answers is not in the document]
Comment (ujjwal-shekhar) [reply hhds/tree.hpp:1224]: The initialization is correct, the variable has been aptly changed to clarify intent. Will be available in the next commit.

**Pull request closed without merging**